### PR TITLE
convert `$` to native JS

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Implements a native Javascript implementation of `$`, which treats the HTML as such and does not insert it as text.

Follow-up on #24049 and proper implementation of f7a6cbd529ee4ad5a843535d8a1f0b036adaa3e1.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

Less reliance on `$`.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Tested

- [X] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/adblock-ask.ts [hunk 1]
```diff
@@ -23,16 +22,12 @@ const canShow = () =>
...
-					slot.append(askHtml);
-				});
-			})
-	);
+	return fastdom
+		.measure(() => document.querySelector('.js-aside-slot-container'))
+		.then((slot) => {
+			if (!slot) return;
+			return fastdom.mutate(() => {
+				slot.insertAdjacentHTML('beforeend', askHtml);
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/adblock-ask.ts:30]: This maintains the original JS logic.

**Comment by mxdvl** [comment]
```
const pre = `acquisitionData=%7B%22componentType%22%3A%22ACQUISITIONS_OTHER%22%2C%22source%22%3A%22GUARDIAN_WEB%22%2C%22campaignCode%22%3A%22shady_pie_open_2019%22%2C%22componentId%22%3A%22shady_pie_open_2019%22%7D&INTCMP=shady_pie_open_2019`

const params = new URLSearchParams();
params.set(
	'acquisitionData',
	JSON.stringify({
		componentType: 'ACQUISITIONS_OTHER',
		source: 'GUARDIAN_WEB',
		campaignCode: 'shady_pie_open_2019',
		componentId: 'shady_pie_open_2019',
	}),
);
params.set('INTCMP', 'shady_pie_open_2019');

const post = params.toString();

console.log(pre === post);

```

**Comment by mxdvl** [comment]
To test 9dc5851:

```js
const pre = `acquisitionData=%7B%22componentType%22%3A%22ACQUISITIONS_OTHER%22%2C%22source%22%3A%22GUARDIAN_WEB%22%2C%22campaignCode%22%3A%22shady_pie_open_2019%22%2C%22componentId%22%3A%22shady_pie_open_2019%22%7D&INTCMP=shady_pie_open_2019`

const params = new URLSearchParams();
params.set(
	'acquisitionData',
	JSON.stringify({
		componentType: 'ACQUISITIONS_OTHER',
		source: 'GUARDIAN_WEB',
		campaignCode: 'shady_pie_open_2019',
		componentId: 'shady_pie_open_2019',
	}),
);
params.set('INTCMP', 'shady_pie_open_2019');

const post = params.toString();

console.log("Are the two string identical?", pre === post ? "Yes, identical!" : "No: different");
```

File: static/src/javascripts/projects/commercial/adblock-ask.ts (hunk 1 above)
Comment (arelra) [reply static/src/javascripts/projects/commercial/adblock-ask.ts:42]: Could we add a test to capture the insertion of this element into the dom and the generated URL?

Comment (mxdvl) [reply static/src/javascripts/projects/commercial/adblock-ask.ts:42]: I was thinking about this and just got a tad lazy :)

**Review by chrislomaxjones — APPROVED** [review]

**Review by arelra — APPROVED** [review]

**Pull request merged**